modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java

```diff
@@ -148,6 +152,39 @@ public String indexName() {
 
     /** {@inheritDoc} */
     @Override public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
-        return super.computeSelfCost(planner, mq).plus(planner.getCostFactory().makeTinyCost());
+        double rows = table.getRowCount();
+
+        double cost = rows * IgniteCost.ROW_PASS_THROUGH_COST;
```

Comment:
u can place this cost computation under "else" scope.

Reply:
yep, but I would prefer to left this as is, If you don't mind